Improve README and explanations (#6)
mesg-foundation/aragon: README.md

# Milestone #1: backend application connecting Aragon DAOs to a webhook connector

## Architecture

```ascii

+------------+                                +-------------+
|            |                                |             |
| Aragon DAO |                                |   Webhook   |
|            |                                |             |
+-----+------+                                +------+------+

>       ^                                              ^

Reviewer: ```suggestion
      |                                              ^
```

Author: I put the arrow in this way to show that the Ethereum service is actively listening for the events. It's a pull and not a push.

      |                                              |
    listen                                        trigger
      |                                              |

>       |                                              |

Reviewer: ```suggestion
      v                                              |
```

+------------+                                +-------------+
|            |                                |             |
|  Ethereum  |                                |   Webhook   |
|  service   |                                |   service   |
|            |                                |             |
+-----+------+                                +------+------+
      |                                              ^
      |                                              |
      |                                              |
    event                                          call
      |             +------------------+             |
      |             |                  |             |
      +------------>+     Process      +-------------+
                    |                  |
                    +------------------+

```

- The Ethereum service source code is available here: https://github.com/mesg-foundation/service-ethereum-contract.
- The Webhook service source code is available here: https://github.com/mesg-foundation/service-webhook

## Step by step explanation

> 1. A MESG Ethereum service is running and actively listens for events from specific Aragon DAO.

Reviewer: ```suggestion
1. A MESG Ethereum service is running and actively listening for events from specific Aragon DAO.
```

Author: @williamzmorgan what do you think?

Reviewer 2: Yes, good change

```

## Download latest Aragon Core Apps abi

**Repo of the Aragon Core Apps:** https://github.com/aragon/aragon-apps

Follow the url of the network of your choice to download the corresponding Aragon Core apps deployment archive:
- **Mainnet network:** https://github.com/aragon/deployments/tree/master/environments/mainnet
- **Rinkeby network:** https://github.com/aragon/deployments/tree/master/environments/rinkeby

When the download is completed, extract the archive, copy the `artifact.json` this repo's folder and rename it with the name of the app (for example `voting.json` for the voting app). Reproduce this step for every Aragon Core Apps.

## Environment file

Let's setup the env file to configure the process and service correctly.

- Copy the `.env.example` to `.env.voting` for voting app.
- Open `.env` and update the values
  - `PROVIDER_ENDPOINT` is the HTTP endpoint of a web3 provider. You can put your own node or just use Infura for test. Make sure to use the same network as your Aragon DAO.
  - `CONTRACT_ADDRESS` is the smart contract address of the a specific app of the DAO. Can be found on the DAO website in `SYSTEM` > `Organization` and `INSTALLED ARAGON APPS`.

>   - `CONTRACT_ABI` is the full ABI of the smart contract. If you have `jq` installed, just update `ARTIFACT_FILE`.

Reviewer: This `jq` part is not clear, I guess should be something like that?
```suggestion
  - `CONTRACT_ABI` is the full ABI of the smart contract. If you have `jq` installed, just run `cat ARTIFACT_FILE | jq .abi`.
```

Author: `cat ARTIFACT_FILE | jq .abi` is already in the .env file and is correctly executed when running `source .env`.
So instead of copy paste the whole ABI in the .env, the dev just needs to replace the ARTICLE_FILE and leave the rest of the command

>   - `CONTRACT_ABI` is the full ABI of the smart contract. If you have `jq` installed, just update `ARTIFACT_FILE`.

Author: ```suggestion
  - `CONTRACT_ABI` is the full ABI of the smart contract. If you have `jq` installed, just update `ARTIFACT_FILE`, otherwise put the whole ABI.
```
